File: db/attachment_type.php
<?php

require_once("entity.php");

/**
 * Class for accessing, manipulating, creating and deleting records in attachment_type.
*/

class Attachment_Type extends Entity
{

  public function __construct($select = array())
  {
    $this->table = "attachment_type";
    $this->order = "";
    $this->domain = "valuelist";
    $this->field['attachment_type_id']['type'] = 'SERIAL';
    $this->field['attachment_type_id']['not_null'] = true;
    $this->field['tag']['type'] = 'VARCHAR';
    $this->field['tag']['length'] = 32;
    $this->field['tag']['not_null'] = true;
    $this->field['rank']['type'] = 'INTEGER';
    $this->field['attachment_type_id']['primary_key'] = true;
    parent::__construct($select);
  }

}

?>

File: pentabarf/lib/momomoto/tables/attachment_type.rb
module Momomoto
  class Attachment_type < Base
    def initialize
      super
      @domain = 'valuelist'
      @fields = {
        :attachment_type_id => Datatype::Integer.new( {:not_null=>true, :default=>true, :primary_key=>true, :serial=>true} ),
        :tag => Datatype::Varchar.new( {:length=>32, :not_null=>true} ),
        :rank => Datatype::Integer.new( {} )
      }
    end
  end
end
